### DISCUSSION TITLE
Promote attributes for user

### DISCUSSION THREAD

**Discussion by ep4sh** [keycloak/keycloak#24326] [Q&A] [answered]
Dear community,

could someone advise me how to promote automatically attributes for the newly created user.

I have Identity provider with user list, and when a user is being created in Keycloak, I want to assign some attributes for it.
Is it possible?

**Comment by danielFesenmeyer** [comment, marked as answer]
That's possible with IDP mappers: https://www.keycloak.org/docs/latest/server_admin/index.html#_mappers

Commonly used is the `Attribute Importer` mapper: It allows to map arbitrary claims to user attributes.
